pkg/utils/parse_arf_result.go

```diff
@@ -348,39 +350,64 @@ func remediationFromFixElement(scheme *runtime.Scheme, fix *xmlquery.Node, scanN
 	dnsFriendlyFixId := strings.ReplaceAll(fixId, "_", "-")
 	remName := fmt.Sprintf("%s-%s", scanName, dnsFriendlyFixId)
 	// TODO(OZZ) fix text
-	return remediationFromString(scheme, remName, namespace, fix.InnerText())
+	return remediationsFromString(scheme, remName, namespace, fix.InnerText())
 }
 
-func remediationFromString(scheme *runtime.Scheme, name string, namespace string, fixContent string) *compv1alpha1.ComplianceRemediation {
-	obj, err := rawObjectToUnstructured(scheme, fixContent)
+func remediationsFromString(scheme *runtime.Scheme, name string, namespace string, fixContent string) []*compv1alpha1.ComplianceRemediation {
+	rems := []*compv1alpha1.ComplianceRemediation{}
+	objs, err := ReadObjectsFromYAML(strings.NewReader(fixContent))
 	if err != nil {
 		return nil
 	}
 
-	annotations := make(map[string]string)
+	for idx := range objs {
+		obj := objs[idx]
+		annotations := make(map[string]string)
 
-	if hasDependencyAnnotation(obj) {
-		annotations = handleDependencyAnnotation(obj)
-	}
+		if hasDependencyAnnotation(obj) {
+			annotations = handleDependencyAnnotation(obj, annotations)
+		}
 
-	return &compv1alpha1.ComplianceRemediation{
-		ObjectMeta: v1.ObjectMeta{
-			Name:        name,
-			Namespace:   namespace,
-			Annotations: annotations,
-		},
-		Spec: compv1alpha1.ComplianceRemediationSpec{
-			ComplianceRemediationSpecMeta: compv1alpha1.ComplianceRemediationSpecMeta{
-				Apply: false,
+		if hasOptionalAnnotation(obj) {
+			annotations = handleOptionalAnnotation(obj, annotations)
+		}
+
+		remType := compv1alpha1.ConfigurationRemediation
+		if hasTypeAnnotation(obj) {
+			anns := obj.GetAnnotations()
+			remType = compv1alpha1.RemediationType(anns[remediationTypeAnnotationKey])
```

Comment:
I wonder if the configuration vs enforcement should be a label?

Reply:
I mean a label in addition to the type. Maybe the type should even be printed in the defaults columns. What I'm getting at is almost everyone will be interested in the configuration remediations, but fewer people will be interested in the enforcement remediations.
btw should we auto-apply also the enforcement remediations?

Reply:
With the current iteration of this, enforcement remediations won't be auto-applied unless you explicitly set that in the ScanSettings. If you set the appropriate type or "all", then we'll do that. What do you think?